### ISSUE TITLE
[core/file_sys/path_parser.cpp] BuildHostPath does not properly take into account different directory seperators.

### ISSUE THREAD

**Issue by wertercatt** [citra-emu/citra#6261]
The bugged function can be found at Line 88.
When using Citra on a Windows machine, it will incorrectly build the path with / directory separators, instead of the \ separators that Windows expects. This can cause it to fail when attempting to read or write to files.
Incorrect behavior:
>[  23.342346] Service.FS <Critical> core/file_sys/archive_sdmc.cpp:OpenFileBase:109: (unreachable) Unknown error opening C:\Users\Kostas\AppData\Roaming/Citra/sdmc/roms/nds/saves/white2.sav

The correct HostPath would be C:\Users\Kostas\AppData\Roaming\Citra\sdmc\roms\nds\saves\white2.sav

**Comment by SachinVin** [comment]
Hi, can you post a log with this build https://github.com/SachinVin/citra/suites/10643663797/artifacts/531732747

**Comment by KostaSaizo7** [comment]
[citra_log.txt](https://github.com/citra-emu/citra/files/10529552/citra_log.txt)

**Comment by SachinVin** [comment]
> [  37.110064] Service.FS <Critical> core/file_sys/archive_sdmc.cpp:OpenFileBase:109: Error opening C:\Users\Kostas\Desktop\head/user/sdmc/roms/nds/saves/white2.sav: The process cannot access the file because it is being used by another process.

**Comment by KostaSaizo7** [comment]
No, I don't even know how to edit a .sav file. I don't have anything open other than citra from the build you posted. To clarify I am using the patch from https://github.com/zaksabeast/DreamRadarCartRedirect/issues/11#issuecomment-1399798079.

**Comment by SachinVin** [comment]
Either it's a bug in the patch, where the save file is reopened without closing it the first time, or with citra (though that might be unlikely because games can open files without issues)

**Comment by zaksabeast** [comment]
Multiple people have used the patch successfully on physical 3ds consoles without issue, and I have been able to use it with citra on a Mac multiple times without issue.  The two people in this thread are using Windows, which is why there are suspicions of the host OS playing a role.

Troubleshooting steps that have been taken:
- Running citra as admin to rule out permission related issues
- Configuring citra to use a desktop directory as the sd card to rule out permission related issues
- Confirming I can use citra on Mac with their same save files to rule out save file issues
- Verifying the save file paths used on the Windows machines to rule out setup issues
- Making sure the actual in-game transfer from Dream Radar to White 2 is being done correctly to rule out usage issues
- Trying multiple versions of citra (I originally tested with a few month old build) to rule out a recently introduced bug

We're at a loss as to what else can be tested at this point.  Given it works on physical 3ds consoles and Mac without issues, and no other causes have been found, it's difficult not to wonder if Windows or something citra does on Windows is at play.

**Comment by SachinVin** [comment]
The log doesn't say that the path is incorrect, only that the file "cannot be accessed because it is being used by another process"(likely citra itself). Windows will lock files for editing by default, while *nix (and the 3DS probably) don't. 
I'm not saying it is not citra's or windows' quirks, but that it is not because of the mismatched `\` and `/`.

**Comment by zaksabeast** [comment]
Yeah, I didn't think the issue was mismatched `\` and `/` either.

Dream Radar checks if a Black 2 or White 2 cart is inserted before sending data to the saves.  If a cart isn't inserted or the cart has the wrong id, Dream Radar shows a message telling the user to insert the correct game.

My patch updates the logic to check if the save file exists on the sd card.  [A comment in the patch repo issue](https://github.com/zaksabeast/DreamRadarCartRedirect/issues/11#issuecomment-1400716372) mentioned moving the file to an incorrect location has Dream Radar act as if a cartridge isn't inserted.  Putting it in the correct place has Dream Radar act as if the cartridge is inserted.

Even without logs, this demonstrated citra was able to find the file without issue.

**Comment by SachinVin** [comment]
I think we should keep this open, because behaviour deviates from actual 3ds

**Comment by KostaSaizo7** [comment]
I think https://github.com/citra-emu/citra/issues/5352 is about actual 3ds behaviour.
